File: Leetcode/HumbleNumbers/209minimum-size-subarray-sum/C++/209minimum-size-subarray-sum.cpp
#include<iostream>
#include<vector>
class Solution {
public:
    int minSubArrayLen(int target, std::vector<int>& nums) {
        int n=nums.size();
        int sum=0;
        int min_length=INT32_MAX;
        int left=0;
        for (int i = 0; i < n; i++)
            {
                /* code */
                sum+=nums[i];
                while(sum>=target)
                {
                   
                       min_length=i-left+1<min_length? i-left+1:min_length;
                       sum-=nums[left];
                        left+=1;              
                }      
            }        
          return min_length==INT32_MAX ? 0:min_length;  
    }
    
};
int main(int argc, char const *argv[])
{
   int target = 4;
    std::vector<int>nums{1,4,4};
    Solution I;
    int &&t=I.minSubArrayLen(target,nums);
    std::cerr<<"t="<<t<<std::endl;
}

File: Leetcode/HumbleNumbers/209minimum-size-subarray-sum/209minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        sum=0
        left=0
        current_length=0
        max_length=0
        arr=[]
        for i in range(0,len(nums)):
            sum+=nums[i];
            current_length+=1
            while sum>=target:
                left+=1
                current_length-=1
                sum-=nums[i]
            if current_length>max_length:max_length=current_length
        return left
def main():
    k=Solution()
    target = 7
    nums = [2,3,1,2,4,3]
    t=k.minSubArrayLen(target,nums)
    print("t="+str(t))
if __name__=="__main__":
    main()
                
                
                
            